pkg/alertmanager/alertmanager.go

```diff
@@ -340,6 +342,27 @@ func (am *Alertmanager) GetReceiversHandler(w http.ResponseWriter, _ *http.Reque
 	}
 }
 
+func (am *Alertmanager) TestReceiversHandler(w http.ResponseWriter, r *http.Request) {
+	c := alertingNotify.TestReceiversConfigBodyParams{}
+	if err := json.NewDecoder(r.Body).Decode(&c); err != nil {
+		http.Error(w,
+			fmt.Sprintf("error unmarshalling test receivers config JSON: %s", err.Error()),
+			http.StatusBadRequest)
+	}
+	response, err := alertingNotify.TestReceivers(r.Context(), c, am.cfg.Templates, am.buildGrafanaReceiverIntegrations, am.cfg.ExternalURL.String())
```

Comment:
I just tried sending a test notification using a custom template but it didn't work. I think this line is the reason why.

The `TestReceivers` function expects the **contents** of the templates, but `am.cfg.Templates` is just the filenames. Check [how `ApplyConfig()` uses this field](https://github.com/grafana/mimir/blob/e32583d81ccf6fba67b230ab7e6c3a2822c973c0/pkg/alertmanager/alertmanager.go#L360-L370), it uses `templates.FromGlobs()` while `TestReceivers` uses `templates.FromContent()`

Reply:
If we got https://github.com/grafana/mimir/pull/4967 merged, then the mess of having to write templates to disk would disappear. WDYT?